bluemira/geometry/parameterisations.py

```diff
@@ -232,6 +234,111 @@ def from_json(cls, file: Union[str, TextIO]) -> GeometryParameterisation:
         var_dict = json.load(file)
         return cls(var_dict)
 
+    def _annotator(self, ax, key: str, xy1: Tuple, xy2: Tuple, xy3: Tuple):
+        """
+        Create annotation arrow with label
+
+        Parameters
+        ----------
+        ax: Axes
+            Matplotlib axis instance
+        key: str
+            label of annotation
+        xy1: Tuple
+            Tuple for first arrow point
+        xy2: Tuple
+            Tuple for second arrow point
+        xy3: Tuple
+            Tuple for arrow label location
+
+        """
+        key = key.replace("_", "")  # TODO proper latex
+        ax.annotate(
+            "",
+            xy=xy1,
+            xycoords="data",
+            xytext=xy2,
+            textcoords="data",
+            arrowprops={
+                "arrowstyle": "<|-|>",
+                "edgecolor": "k",
+                "facecolor": "k",
+                "shrinkA": 0,
+                "shrinkB": 0,
+            },
+        )
+        ax.annotate(
+            r"$\it{" f"{key}" "}$",
+            xy=xy3,
+            xycoords="data",
+            xytext=(0, 5),
+            textcoords="offset points",
+        )
+
+    def _label_function(self, ax, shape):
+        """
+        Adds labels to parameterisation plots
+
+        Parameters
+        ----------
+        ax: Axes
+            Matplotlib axis instance
+        shape: BluemiraWire
+            parameterisation wire
+
+        """
+        offset_ar_x = 0
+        offset_ar_z = 0
+        for key, var in self.variables.as_dict().items():
+            if key.startswith("x"):
+                self._annotator(
+                    ax,
+                    key,
+                    (0, offset_ar_x),
+                    (var["_value"], offset_ar_x),
+                    (var["_value"] * 0.4, offset_ar_x),
+                )
+                ax.plot([0, 0], [0, offset_ar_x], color="k")
+                ax.plot([var["_value"], var["_value"]], [0, offset_ar_x], color="k")
+                offset_ar_x += 2
+            elif key.startswith("z") or key[1] == "z":
+                xcor = shape.center_of_mass[0] + offset_ar_z
+                self._annotator(
+                    ax,
+                    key,
+                    (xcor, 0),
+                    (xcor, var["_value"]),
+                    (xcor, var["_value"] * 0.4),
+                )
+                ax.plot([shape.center_of_mass[0], xcor], [0, 0], color="k")
+                ax.plot(
+                    [shape.center_of_mass[0], xcor],
+                    [var["_value"], var["_value"]],
+                    color="k",
+                )
+                offset_ar_z += 1.5
+        return offset_ar_x, offset_ar_z
+
+    def plot(self, ax=None, labels=True, **kwargs):
```

Comment:
`kwargs` here are unused. Should they be passed into the `plot_2d` call?

Reply:
Should we be plotting labels by default? Feels to me like it's something users would use as a one-off, but wouldn't be using in the general case

Reply:
Done on both counts